external-api/src/repositories/estore/estore-repo.ts

```diff
@@ -0,0 +1,80 @@
+import { InsertOneResult, ObjectId, UpdateFilter, UpdateResult, WithId, WithoutId } from 'mongodb';
+import { MONGO_DB_COLLECTIONS } from '@ukef/dtfs2-common';
+import { getCollection } from '../../database';
+
+/**
+ * Finds a document in the CRON_JOB_LOGS collection based on the provided deal ID.
+ *
+ * @param {ObjectId} dealId - The unique identifier for the deal.
```

Comment:
Sorry I didn't pick up on this last time, but we shouldn't have types in both the TSDoc and the function signature

Reply:
I replied back to your previous comment regarding the same, we are still using types in arguments (JSDocs - https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#param-and-returns) across the repo I would unify this in an epic.

Signature is always a good to specify explicitly, albeit not mandatory.

Reply:
I think I'm going to push back on this a bit simply citing something that has already happened below in this PR - you updated the return type of this function but didn't have to update the return type in the jsdoc and they're now out of sync.

The link you provided doesn't actually say anything about providing both jsdoc types and typescript types - it is discussing the fact that you _can_ use the typescript type checker with jsdoc inside of js files, which is useful if you set the tsconfig `checkJs` option to `true`. However, minimising confusion by having one source of truth with respect to types is quite important in my opinion, and avoiding any risk of confusing the LSP by having two sources for type definitions is also important for the developer experience

Reply:
I didn't see the reply to the previous comment. We are likely to have a code base which forever uses a mix of JSDoc and TSDoc, so using TSDoc in typescript files and JSDoc in javascript files seems like the standard we should stick to for future code

Reply:
Previous response - https://github.com/UK-Export-Finance/dtfs2/pull/3390/files/2f0284dbe0e71d65cdf67c321a8d47d23f42cd3f#r1722864347

* I have updated the return signature which I should have updated, however rightly so this has been picked up in a review.
* Majority of the TS files still uses JSDocs and until a convention is decided to migrate all TS files to use TSDocs.

Reply:
Form a quick search it appears that this is not the norm in most typescript files and is instead simply a common occurrence in `external-api` typescript files. I would prioritise fixing this issue in `external-api` sooner rather than later as continuing to add JSDoc types in typescript files will just add to the pile of tech debt and discourage performing this refactor in future. If there is a ticket number related to updating this, can we please add it as a TODO and then I'll be happy to resolve this comment thread

Reply:
I am happy add an existing ticket number to the `README.md` rather than doc string as this affects all the micro-services including existing PRs.

Reply:
If you can please add this as a TODO in the `README` then I'm happy to approve

Reply:
Thanks @JoshBinns2000  have added to a new `docs/docstring.md` file.